src/sentry/static/sentry/app/views/settings/project/projectFilters/index.jsx

```diff
@@ -52,34 +40,30 @@ class ProjectFilters extends React.Component {
               className="nav nav-tabs"
               style={{borderBottom: '1px solid #ddd', paddingTop: '30px'}}
             >
-              <li className={activeSection === 'data-filters' ? 'active' : ''}>
-                <a onClick={() => this.setProjectNavSection('data-filters')}>
+              <li className={filterType === 'data-filters' ? 'active' : ''}>
+                <Link to={recreateRoute('data-filters/', {...this.props, stepBack: -1})}>
                   {t('Data Filters')}
-                </a>
+                </Link>
               </li>
-              <li className={activeSection === 'discarded-groups' ? 'active' : ''}>
-                <a onClick={() => this.setProjectNavSection('discarded-groups')}>
+              <li className={filterType === 'discarded-groups' ? 'active' : ''}>
+                <Link
+                  to={recreateRoute('discarded-groups/', {...this.props, stepBack: -1})}
```

Comment:
Welp, subroute works better when we don't have this `recreateRoute` multiple route situation :(